### PR TITLE
[Lib][Aggregator] Support inconsecutive process id.

### LINKED ISSUE #225: Support inconsecutive id in AggregatorFactoryBase
#173 supports inconsecutive id for process.
In https://github.com/husky-team/husky/blob/master/lib/aggregator.cpp#L250,  it still relies on the consecutiveness.

### PR DESCRIPTION
**Pull request by zzxx-husky**
Fix #225

### CODE REVIEW

File: lib/aggregator.cpp [hunk 1]
```diff
@@ -212,8 +212,8 @@ void AggregatorFactoryBase::synchronize() {
 bool AggregatorFactoryBase::same_machine(size_t fid) { return get_machine_id() == get_machine_id(fid); }
 
 size_t AggregatorFactoryBase::get_local_center(size_t machine_id, size_t agg_idx) {
-    auto& all_factory = shared_data_->all_factory[machine_id];
-    return all_factory[agg_idx % all_factory.size()];
+    auto& factory = shared_data_->all_factory[machine_id];
```
Comment (kygx-legend) [inline lib/aggregator.cpp]: Seems it needs a safe check in `all_factory[machine_id]`?

File: lib/aggregator.cpp [hunk 2]
```diff
@@ -245,23 +245,19 @@ void AggregatorFactoryBase::InnerSharedData::initialize(AggregatorFactoryBase* f
...
     size_t num_global_factory = factory->get_num_global_factory();
     num_holder.store(static_cast<uint32_t>(num_local_factory));
-    std::vector<std::vector<size_t>>& fids = all_factory;
-    fids.resize(factory->get_num_machine());
-    for (size_t i = 0; i < num_global_factory; i++) {
-        fids[factory->get_machine_id(i)].push_back(i);
-    }
-    local_centers = fids[factory->get_machine_id()];
+    all_factory = factory->get_all_factory();
+    local_centers = all_factory[factory->get_machine_id()];
```
Comment (kygx-legend) [inline lib/aggregator.cpp]: Ditto.

File: lib/aggregator.cpp (hunk 1 above)
Comment (zzxx-husky) [inline lib/aggregator.cpp]: 0.0 I replaced that with `std::unordered_map::at`

Comment (kygx-legend) [inline lib/aggregator.cpp]: Cool

**Pull request merged**